config/helpers/bootstrap.go

```diff
@@ -261,22 +262,39 @@ func BindRootAPIs(ctx context.Context, kcpClient kcpclient.Interface, exportName
 
 		created, err := kcpClient.ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
 		if err == nil {
-			klog.V(2).Infof("Created API binding %s|%s", logicalcluster.From(created), created.Name)
+			logger := logging.WithObject(logger, created)
+			logger.V(2).Info("Created API binding")
 			continue
 		}
 		if !apierrors.IsAlreadyExists(err) {
 			return err
 		}
 
-		existing, err := kcpClient.ApisV1alpha1().APIBindings().Get(ctx, exportName, metav1.GetOptions{})
-		if err != nil {
+		if err := wait.PollImmediateInfiniteWithContext(ctx, time.Second, func(ctx context.Context) (bool, error) {
+			existing, err := kcpClient.ApisV1alpha1().APIBindings().Get(ctx, exportName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
```

Comment:
Why is it ok for this to return `err` and fail?

Reply:
You're right, this hides the same issue I am trying to fix. If we get an already-exists error trying to create, and then the get request fails, we're probably in some weird perma-fail state at that point. Could log + return `false, nil`, could panic...?

Reply:
Updated so we never return an error and always keep retrying